src/restconfig-wmts/src/test/java/org/geoserver/rest/service/LocalWMTSSettingsControllerTest.java

```diff
@@ -0,0 +1,222 @@
+                        json,
+                        "text/json");
+        assertEquals(200, response.getStatus());
+        JSON jsonMod =
+                getAsJSON(
+                        RestBaseController.ROOT_PATH
+                                + "/services/wmts/workspaces/sf/settings.json");
+        JSONObject jsonObject = (JSONObject) jsonMod;
+        assertNotNull(jsonObject);
+        JSONObject wmtsinfo = (JSONObject) jsonObject.get("wmts");
+        // assertEquals("wmts", wmtsinfo.get("id"));
+        assertEquals("false", wmtsinfo.get("enabled").toString().trim());
+    }
+
+    @Test
+    public void testPutAsXML() throws Exception {
+        String xml =
+                "<wmts>"
+                        + "<id>wmts</id>"
+                        + "<workspace>"
+                        + "<name>sf</name>"
+                        + "</workspace>"
+                        + "<enabled>false</enabled>"
+                        + "</wmts>";
+        MockHttpServletResponse response =
+                putAsServletResponse(
+                        RestBaseController.ROOT_PATH + "/services/wmts/workspaces/sf/settings",
+                        xml,
+                        "text/xml");
+        assertEquals(200, response.getStatus());
+        Document dom =
+                getAsDOM(
+                        RestBaseController.ROOT_PATH + "/services/wmts/workspaces/sf/settings.xml");
+        assertXpathEvaluatesTo("false", "/wmts/enabled", dom);
+    }
+
+    @Test
+    public void testPutFullAsXML() throws Exception {
+        String xml =
+                IOUtils.toString(
+                        LocalWMTSSettingsControllerTest.class.getResourceAsStream("wmts.xml"),
+                        "UTF-8");
+        MockHttpServletResponse response =
+                putAsServletResponse(
+                        RestBaseController.ROOT_PATH + "/services/wmts/workspaces/sf/settings",
+                        xml,
+                        "text/xml");
+        assertEquals(200, response.getStatus());
+        Document dom =
+                getAsDOM(
+                        RestBaseController.ROOT_PATH + "/services/wmts/workspaces/sf/settings.xml");
+        assertXpathEvaluatesTo("true", "/wmts/enabled", dom);
+    }
+
+    @Test
+    public void testDelete() throws Exception {
+        assertEquals(
+                200,
+                deleteAsServletResponse(
+                                RestBaseController.ROOT_PATH
+                                        + "/services/wmts/workspaces/sf/settings")
+                        .getStatus());
+        boolean thrown = false;
+        try {
+            JSON json =
+                    getAsJSON(
+                            RestBaseController.ROOT_PATH
+                                    + "/services/wmts/workspaces/sf/settings.json");
+        } catch (JSONException e) {
+            thrown = true;
+        }
```

Comment:
For a shorter test consider:
```
@Test(expected = JSONException.class)
```

Reply:
right, mind, it's a copy of the other module... maybe we could do some general cleanup but not as part of this pr? do you see this as a blocker?